pkg/status/statuscollector.go

```diff
@@ -50,3 +65,92 @@ func (c *Controller) syncStatusCollector(ctx context.Context, ref string) error
 	logger.Info("Synced StatusCollector", "ref", ref)
 	return nil
 }
+
+func (c *Controller) validateStatusCollector(statusCollector *v1alpha1.StatusCollector) []error {
+	var errs []error
+	// groupBy & CombinedFields empty if select is not
```

Comment:
Wouldn't it be better to inject these empty/non-empty constraints into the OpenAPI schema?

Reply:
I gave it a go and it did not work out as expected, I rather not handle this now. Can open issue on using openapi validations where makes sense across our API CRDs.

Reply:
OK